parity-crypto/src/publickey/secret_key.rs

```diff
@@ -206,14 +208,18 @@ impl FromStr for Secret {
 }
 
 impl From<[u8; 32]> for Secret {
-	fn from(k: [u8; 32]) -> Self {
-		Secret { inner: H256(k) }
+	fn from(mut k: [u8; 32]) -> Self {
+		let result = Secret { inner: Box::new(H256(k)) };
```

Comment:
Dumb q: when we create a `Box(H256(k))`, does it first create `H256(k)` on the stack and then copies it to the heap?
If it does, should we use something like https://github.com/kvark/copyless or unsafe to avoid it?

Reply:
I did not see copy of k to create h256 in debug from a test case, but from calling point it is easy to have some (eg using into instead of from), but I hope things would behave better when optimized.